File: app/app/src/main/java/com/cabal/app/navigation_bar/MyEventsFragment.java
package com.cabal.app.navigation_bar;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.cabal.app.AddEventFragment;
import com.cabal.app.MyEventsAdapter;
import com.cabal.app.R;
import com.cabal.app.Utils.Filters;
import com.cabal.app.Utils.JsonLoader;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.util.Objects;

public class MyEventsFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_my_events, container, false);
        FloatingActionButton btnAddEvent = view.findViewById(R.id.btnAddEvent);
        CheckBox authorFilter = view.findViewById(R.id.authorSwitch);
        CheckBox guestFilter = view.findViewById(R.id.guestSwitch);
        CheckBox finishedFilter = view.findViewById(R.id.finishedSwitch);
        MyEventsAdapter adapter = new MyEventsAdapter(JsonLoader.INSTANCE.loadEvents(view.getContext()));
        authorFilter.setOnCheckedChangeListener((v, checked) -> {
            Filters.INSTANCE.setAUTHOR_FILTER(checked);
            adapter.getFilter().filter("");
        });
        guestFilter.setOnCheckedChangeListener((v, checked) -> {
            Filters.INSTANCE.setGUEST_FILTER(checked);
            adapter.getFilter().filter("");
        });
        finishedFilter.setOnCheckedChangeListener((v, checked) -> {
            Filters.INSTANCE.setFINISHED_FILTER(checked);
            adapter.getFilter().filter("");
        });
        btnAddEvent.setOnClickListener(view1 -> Objects.requireNonNull(getActivity())
                .getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.fragment_container, new AddEventFragment())
                .addToBackStack(null)
                .commit());
        RecyclerView recyclerView = view.findViewById(R.id.recycleMyEvents);
        recyclerView.setLayoutManager(new LinearLayoutManager(view.getContext()));
        recyclerView.setAdapter(adapter);
        return view;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }
}

File: app/app/src/main/java/com/cabal/app/my_events_mvvm/MyEventsFragment.kt
package com.cabal.app.my_events_mvvm

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.cabal.app.AddEventFragment
import com.cabal.app.MyApplication
import com.cabal.app.R
import com.google.android.material.floatingactionbutton.FloatingActionButton
import javax.inject.Inject

class MyEventsFragment : Fragment() {

    @Inject
    lateinit var factory: ViewModelProvider.Factory


    private lateinit var viewModel: MyEventsViewModel

    private lateinit var adapter: MyEventsAdapter

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val application = (activity?.application as MyApplication)
        application.appComponent.inject(this)
        adapter = MyEventsAdapter(ArrayList(), application)
        viewModel = ViewModelProviders.of(this, factory)[MyEventsViewModel::class.java]
        viewModel.events.observe(this, Observer {
            adapter.filteredEvents = it
            adapter.originalEvents = it
            adapter.notifyDataSetChanged()
        })

        val view = inflater.inflate(R.layout.fragment_my_events, container, false)
        view.findViewById<FloatingActionButton>(R.id.btnAddEvent).setOnClickListener {
                        activity!!.supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.fragment_container, AddEventFragment())
                    .addToBackStack(null)
                    .commit()
        }

        view.findViewById<CheckBox>(R.id.authorSwitch).setOnCheckedChangeListener { _, checked ->
            adapter.filters.AUTHOR_FILTER = checked
            adapter.filter.filter("")
        }
        view.findViewById<CheckBox>(R.id.guestSwitch).setOnCheckedChangeListener { _, checked ->
            adapter.filters.GUEST_FILTER = checked
            adapter.filter.filter("")
        }
        view.findViewById<CheckBox>(R.id.finishedSwitch).setOnCheckedChangeListener { _, checked ->
            adapter.filters.FINISHED_FILTER = checked
            adapter.filter.filter("")
        }
        view!!.findViewById<RecyclerView>(R.id.recycleMyEvents).also {
            it.layoutManager = LinearLayoutManager(context)
            it.adapter = adapter
        }
        return view
    }
}